### PR TITLE
EVA-3590 update initial check for validation required or not

### PR DESCRIPTION
**Pull request by nitin-ebi**

### CODE REVIEW

File: eva_sub_cli/main.py [hunk 1]
```diff
@@ -110,6 +114,28 @@ def get_project_and_vcf_fasta_mapping_from_metadata_xlsx(metadata_xlsx, mapping_
...
+                        return True
+                    else:
+                        return False
+                except requests.HTTPError as ex:
+                    logger.error(f'Error occurred while getting status of the submission with ID {submission_id}: {ex.response.status_code} {ex.response.text}')
+                    raise
+            else:
+                logger.info(f'submission id not found. This might be the first time user is submitting')
+                return False
+
```
Comment (apriltuesday) [inline eva_sub_cli/main.py:137]: I'd add a `return False` at the very end here, just to be explicit

**Review by apriltuesday — APPROVED** [review]
I've put some suggestions but overall this looks fine.

File: eva_sub_cli/main.py [hunk 2]
```diff
@@ -110,6 +114,28 @@ def get_project_and_vcf_fasta_mapping_from_metadata_xlsx(metadata_xlsx, mapping_
...
+            if submission_id:
+                try:
+                    submission_status = get_submission_status(submission_id)
+                    if submission_status == 'FAILED':
+                        return True
+                    else:
+                        return False
+                except requests.HTTPError as ex:
+                    logger.error(f'Error occurred while getting status of the submission with ID {submission_id}: {ex.response.status_code} {ex.response.text}')
+                    raise
```
[the remark this answers is not in the document]
Comment (tcezard) [reply eva_sub_cli/main.py:133]: I see two options: 
 - Have a sensible default when the API fails to respond ie submission does not exist
 - Raising an error that should be caught and exit gracefully with a clear error message

**Review by tcezard — APPROVED** [review]
Approach looks good to me. 
Couple of cosmetic points

File: eva_sub_cli/main.py (hunk 1 above)
Comment (nitin-ebi) [reply eva_sub_cli/main.py:143]: updated

File: eva_sub_cli/main.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/main.py]: raising and existing gracefully with msg

File: eva_sub_cli/api_utils.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+def get_submission_uploaded_url(submission_id, sub_ws_url=None):
+    return os.path.join(get_submission_ws_url(sub_ws_url), SUBMISSION_UPLOADED_PATH.format(submissionId=submission_id))
+
+
+def get_submission_status_url(submission_id, sub_ws_url=None):
+    return os.path.join(get_submission_ws_url(sub_ws_url), SUBMISSION_STATUS_PATH.format(submissionId=submission_id))
+
+
+def get_submission_status(submission_id, sub_ws_url=None):
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/api_utils.py]: Added retry. I have tried to implement the 2nd approach suggested

File: eva_sub_cli/main.py [hunk 4]
```diff
@@ -110,6 +114,28 @@ def get_project_and_vcf_fasta_mapping_from_metadata_xlsx(metadata_xlsx, mapping_
     return project_title, vcf_fasta_report_mapping
 
 
+def check_validation_required(tasks, sub_config):
+    # Validation is mandatory so if submit is requested then VALIDATE must have run before or be requested as well
+    if SUBMIT in tasks:
+        if not sub_config.get(READY_FOR_SUBMISSION_TO_EVA, False):
+            return True
+        else:
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/main.py]: updated

File: eva_sub_cli/main.py [hunk 5]
```diff
@@ -110,6 +114,28 @@ def get_project_and_vcf_fasta_mapping_from_metadata_xlsx(metadata_xlsx, mapping_
...
+                try:
+                    submission_status = get_submission_status(submission_id)
+                    if submission_status == 'FAILED':
+                        return True
+                    else:
+                        return False
+                except requests.HTTPError as ex:
+                    logger.error(f'Error occurred while getting status of the submission with ID {submission_id}: {ex.response.status_code} {ex.response.text}')
+                    raise
+            else:
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/main.py]: updated

**Review by apriltuesday — APPROVED** [review]

File: eva_sub_cli/submission_ws.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+        self.base_url = self._submission_ws_url
+
+    SUBMISSION_WS_URL = 'https://www.ebi.ac.uk/eva/webservices/submission-ws/v1/'
+    SUBMISSION_INITIATE_PATH = 'submission/initiate'
+    SUBMISSION_UPLOADED_PATH = 'submission/{submissionId}/uploaded'
+    SUBMISSION_STATUS_PATH = 'submission/{submissionId}/status'
+
+    @property
+    def _submission_ws_url(self):
+        """Retrieve the base URL for the submission web services. In order of preference from the user of this class,
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply eva_sub_cli/submission_ws.py]: updated

File: bin/eva-sub-cli.py [hunk 7]
```diff
@@ -83,3 +86,7 @@ def get_version():
         main.orchestrate_process(**args.__dict__)
     except FileNotFoundError as fne:
         print(fne)
+    except SubmissionNotFoundException as snfe:
+        print(f'{snfe}. Retry with correct submission id. If the problem persists, please contact EVA Helpdesk')
```
[the remark this answers is not in the document]
Comment (nitin-ebi) [reply bin/eva-sub-cli.py]: updated

**Pull request merged**